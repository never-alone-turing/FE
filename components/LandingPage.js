import React, {useState} from "react";
import { View, Text, StyleSheet, Button, FlatList, TouchableOpacity } from "react-native";
// import { NativeRouter, Switch, Route } from 'react-router-native';
import ListItem from './ListItem';


// const LandingPage = ({ history }) => {
//   const [titles, chooseTitle] = useState([
//     {title: "Caretaker", path: '/caretaker'},
//     {title: "Care Receiver", path: '/carereceiver'}
//   ])
//   return (
//     <>
//       <Text style={styles.text}>Never Alone</Text>
//         <FlatList
//           style={styles.accountButton}
//           data={titles}
//           keyExtractor={(item) => item.title}
//           renderItem={({ item }) =>
//             <ListItem
//               item={item} history={history} />}
//         />
//     </>
//   );
// }

const LandingPage = ({ navigation }) => {
  return (
    <View>
      <Text style={styles.text}>Never Alone</Text>
      <TouchableOpacity
        style={styles.careTakerButton}
        onPress={() => navigation.navigate('Caretaker')}
      >
        <Text style={styles.buttonText}>Caretaker</Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.receiverButton}
        onPress={() => navigation.navigate('Receiver')}
      >
        <Text style={styles.buttonText}>Receiver</Text>
      </TouchableOpacity>
    </View>

  )
}

const styles = StyleSheet.create({
  text: {
   color: 'darkslateblue',
   fontSize: 30,
   alignSelf: "stretch",
   textAlign: "center",
   margin: 45
    // backgroundColor: 'green'
  },

  careTakerButton: {
   color: 'black',
   backgroundColor: 'darkslateblue',
   borderColor: 'black',
   borderRadius: 50,
   margin: 30,
   padding: 10,
   fontSize: 30
 },

 receiverButton: {
  color: 'black',
  backgroundColor: 'darkslateblue',
  borderColor: 'black',
  borderRadius: 50,
  margin: 30,
  padding: 10,
  fontSize: 30
 },

 buttonText: {
   color: 'darkgrey',
   fontSize: 30,
   alignSelf: "stretch",
   textAlign: "center",
   margin: 45
 }
})

export default LandingPage;
