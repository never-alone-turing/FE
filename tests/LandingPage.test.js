import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import LandingPage from '../components/LandingPage';

describe('LandingPage', () => {
  it('should match the snapshot', () => {
    const landingpage = renderer.create(<LandingPage/>).toJSON();
    expect(landingpage).toMatchSnapshot()
  });

  // it('should be able to switch pages', () => {
  //   let mockNav;
  //   mockNav.navigate = jest.fn()
  //   const { getByTestId } = render(<LandingPage navigation={mockNav}/>);
  //   const caretakerButton = getByTestId("lp-caretaker-button")
  //   fireEvent.press(caretakerButton)
  //   expect(mockNav).toHaveBeenCalled()
  // });

})

// test('Render LandingPage Properly', () => {
//   const {debug} = render(<LandingPage/>);
//   debug();
// });