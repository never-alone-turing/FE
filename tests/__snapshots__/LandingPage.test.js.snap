// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  testID="lp-view"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "paddingBottom": 10,
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 3,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#fff",
              "fontSize": 60,
              "fontWeight": "bold",
            }
          }
          text="Never Alone"
        >
          Never Alone
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
          }
        }
      />
    </RNCSafeAreaView>
  </View>
  <View
    style={Object {}}
  >
    <Text
      style={
        Object {
          "alignSelf": "stretch",
          "color": "black",
          "fontSize": 60,
          "margin": 90,
          "textAlign": "center",
        }
      }
    >
      Choose Your Role:
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#70cfff",
          "borderColor": "black",
          "borderRadius": 20,
          "color": "black",
          "elevation": 8,
          "fontSize": 50,
          "margin": 40,
          "opacity": 1,
          "padding": 20,
        }
      }
      testID="lp-caretaker-button"
    >
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "color": "black",
            "fontSize": 100,
            "margin": 40,
            "textAlign": "center",
          }
        }
      >
        Caretaker
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#9192ca",
          "borderColor": "black",
          "borderRadius": 20,
          "color": "black",
          "elevation": 8,
          "fontSize": 50,
          "margin": 40,
          "opacity": 1,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "color": "black",
            "fontSize": 100,
            "margin": 40,
            "textAlign": "center",
          }
        }
      >
        Receiver
      </Text>
    </View>
  </View>
</View>
`;
